classes/i18n/LocaleConversion.php

```diff
@@ -23,29 +23,76 @@
 
 namespace PKP\i18n;
 
+use Exception;
 use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Cache;
+use PKP\core\Core;
 use PKP\facades\Locale;
 
 class LocaleConversion
 {
+    /** @var string Max lifetime for the ISO639-2b cache. */
+    protected const MAX_ISO6392b_CACHE_LIFETIME = '1 year';
+
+    /**
+     * A post-install-cmd custom composer script that checks if
+     * the iso_639-2b.json file exists in the installed sokil library
+     *
+     * @throw Exception
+     */
+    public static function composerIsoFileCheckScript(): void
+    {
+        // We use dirname(__FILE__, 3) and not Core::getBaseDir() because
+        // this funciton is called by Composer, where INDEX_FILE_LOCATION is not defined.
+        $iso6392bFile = dirname(__FILE__, 3) . '/lib/vendor/sokil/php-isocodes-db-i18n/databases/iso_639-2.json';
+        if (!file_exists($iso6392bFile)) {
+            throw new Exception("The ISO639-2b file {$iso6392bFile} does not exist.");
+        }
+    }
+
+    /**
+     * Get ISO639-2b array
+     */
+    protected static function getISO6392b(): array
+    {
+        $iso6392bFile = Core::getBaseDir() . '/' . PKP_LIB_PATH . '/lib/vendor/sokil/php-isocodes-db-i18n/databases/iso_639-2.json';
+        if (!file_exists($iso6392bFile)) {
+            error_log("The ISO639-2b file {$iso6392bFile} does not exist.");
+            return [];
```

Comment:
Is this feature essential or cause strange issues if non-functional? If yes, then an exception sounds better.